### PR TITLE
mitmproxy: 5.3.0 -> 6.0.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Success of #110091

- Update to 6.0.2
- Limit to newer Python releases
- Sync requirements to upstream for 6.x
- Switch to `pytestCheckHook`
- Formatting with `nixpkgs-fmt`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113948` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mitmproxy (python38Packages.mitmproxy)</li>
    <li>python39Packages.mitmproxy</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113948` at b022c9f1 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/17eac8b4ba26394b9cd938f80108ef36">mitmproxy</a></li> (python38Packages.mitmproxy)</li>
    <li><a href="https://gist.github.com/2ca5b4847a65ed50361a0b5e87119b9a">python39Packages.mitmproxy</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/mitmproxy/default.nix [hunk 1]
```diff
@@ -96,21 +87,39 @@ buildPythonPackage rec {
...
     parver
-    pytest
     pytest-asyncio
+    pytest-timeout
+    pytest-xdist
+    pytestCheckHook
     requests
   ];
 
+  doCheck = (!stdenv.isDarwin);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mitmproxy/default.nix:101]:
```suggestion
  doCheck = !stdenv.isDarwin;
```

**Review by SuperSandro2000 — DISMISSED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113948` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>mitmproxy (python38Packages.mitmproxy)</li>
<li>python39Packages.mitmproxy</li>
</ul>
</details>

File: pkgs/development/python-modules/mitmproxy/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/mitmproxy/default.nix]: Thanks, updated.

File: pkgs/development/python-modules/mitmproxy/default.nix [hunk 2]
```diff
@@ -96,21 +87,39 @@ buildPythonPackage rec {
...
     pytest-asyncio
+    pytest-timeout
+    pytest-xdist
+    pytestCheckHook
     requests
   ];
 
+  doCheck = !stdenv.isDarwin;
+
+  postPatch = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/mitmproxy/default.nix:103]: Please don't move pieces of code around when there is no need to.

**Review by FRidh — COMMENTED** [review]
Looks good to me, functionally.

File: pkgs/development/python-modules/mitmproxy/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/mitmproxy/default.nix:103]: `nixpkgs-hammer` has complained about the ordering.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/mitmproxy/default.nix:103]: Fine for me when we change bigger parts anyway.

**Pull request merged**